R/uniDimBayesianOutputPlots.R

```diff
@@ -0,0 +1,389 @@
+  lower <- (1-int)/2
+  upper <- int + (1-int)/2
+
+  samp_tmp <- as.vector(coefScale[["samp"]])
+  dat <- data.frame(as.matrix(samp_tmp), row.names =  NULL)
+  names(dat) <- "value"
+  dat$colos <- "1"
+  dat$var <- "original"
+
+  item_tmp <- apply(coefItem[["itemSamp"]], 3, as.vector)
+  dat_del <- t(as.matrix(as.data.frame(item_tmp)))
+  names <- decodeColNames(variables)
+
+  for (i in n_row:1){
+    tmp <- as.data.frame(dat_del[i, ])
+    colnames(tmp) <- "value"
+    tmp$var <- names[i]
+    tmp$colos <- "2"
+    dat <- rbind(dat, tmp)
+  }
+  dat$var <- factor(dat$var, levels = unique(dat$var))
+
+  if (!is.null(ordering)) {
+    est <- as.data.frame(coefItem[["itemEst"]])
+    est[n_row + 1, ] <- 1
+    colnames(est) <- "value"
+    est$name <- c(names, "original")
+
+    if (ordering == "orderItemMean") {
+      dists <- abs(coefScale[["est"]] - coefItem[["itemEst"]])
+      dists[length(dists)+1] <- 0
+      est <- est[order(dists, decreasing = F), ]
+      dat$var <- factor(dat$var, levels = c(est$name))
+
+    } else if (ordering == "orderItemKL") {
+      samps <- item_tmp
+      og_samp <- samp_tmp
+      dists <- apply(samps, 2, .KLD.statistic, y = og_samp) # kl divergence
+      dists[length(dists)+1] <- 0
+      est <- est[order(dists), ]
+      dat$var <- factor(dat$var, levels = c(est$name))
+
+    } else if (ordering == "orderItemKS") {
+      samps <- item_tmp
+      og_samp <- samp_tmp
+      dists <- apply(samps, 2, .ks.test.statistic, y = og_samp) # ks distance
+      dists[length(dists)+1] <- 0
+      est <- est[order(dists), ]
+      dat$var <- factor(dat$var, levels = c(est$name))
+    }
+  }
+
+  g <- ggplot2::ggplot(dat, ggplot2::aes(x = value, y = var, fill = colos)) +
+    ggridges::stat_density_ridges(quantile_lines = T, quantiles = c(lower, 0.5, upper),
+                                  alpha = .85, show.legend = F, scale = 1) +
+    ggplot2::theme(strip.background = ggplot2::element_rect(fill = "white"),
+                   strip.text = ggplot2::element_text(colour = "black")) +
+    ggplot2::ylab(gettext("Item Dropped")) +
+    ggplot2::xlab(nms) +
+    ggplot2::scale_fill_grey() +
+    ggplot2::scale_y_discrete(expand = ggplot2::expand_scale(mult = c(0.1, 0.25)))
+
+  return(jaspGraphs::themeJasp(g))
+
+}
+
+
+
+.BayesianPosteriorPredictive <- function(jaspResults, model, options) {
+
+  if (!is.null(.getStateContainerB(jaspResults)[["omegaPPC"]]$object))
+    return()
+
+  if (options[["dispPPC"]] && options[["omegaScale"]] && is.null(model[["empty"]])) {
+
+    ll <- model[["omegaScale"]][["loadings"]]
+    rr <- model[["omegaScale"]][["residuals"]]
+    cimpl <- ll %*% t(ll) + diag(rr)
+    cobs <- model[["data_cov"]]
+    k <- ncol(cobs)
+    eframe <- data.frame(number = seq(1, k), eigen_value = eigen(cobs)$values)
+    ee_impl <- matrix(0, 1e3, k)
+    for (i in 1:1e3) {
+      dtmp <- MASS::mvrnorm(model[["n"]], rep(0, k), cimpl)
+      ee_impl[i, ] <- eigen(cov(dtmp))$values
```

Comment:
Since there's no need to compute the eigenvectors:
```suggestion
      ee_impl[i, ] <- eigen(cov(dtmp), only.values = TRUE)$values
```